main/src/test/scala/ParseKey.scala

```diff
@@ -8,166 +8,114 @@
 package sbt
 
 import Def.{ displayFull, displayMasked, ScopedKey }
-import sbt.internal.{ TestBuild, Resolve }
-import TestBuild._
-import sbt.internal.util.complete._
+import sbt.internal.{ TestBuild, Resolve }, TestBuild._
+import sbt.internal.util.complete.Parser
 
-import org.scalacheck._
-import Gen._
-import Prop._
-import Arbitrary.arbBool
+import org.scalacheck._, Arbitrary.arbitrary, Gen._, Prop._
 
 /**
  * Tests that the scoped key parser in Act can correctly parse a ScopedKey converted by Def.show*Key.
  * This includes properly resolving omitted components.
  */
 object ParseKey extends Properties("Key parser test") {
-  final val MaxKeys = 5
-  final val MaxScopedKeys = 100
-
-  implicit val gstructure = genStructure
-
-  property("An explicitly specified axis is always parsed to that explicit value") =
-    forAllNoShrink(structureDefinedKey) { (skm: StructureKeyMask) =>
-      import skm.{ structure, key, mask => mask0 }
-
+  property("An explicitly specified axis is always parsed to that explicit value") = forAll {
+    (skm: StructureKeyMask) =>
+      import skm.{ structure, key }
       val hasZeroConfig = key.scope.config == Zero
-      val mask = if (hasZeroConfig) mask0.copy(project = true) else mask0
-      val expected = resolve(structure, key, mask)
+      val mask = if (hasZeroConfig) skm.mask.copy(project = true) else skm.mask
       // Note that this explicitly displays the configuration axis set to Zero.
       // This is to disambiguate `proj/Zero/name`, which could render potentially
-      // as `Zero/name`, but could be interpretted as `Zero/Zero/name`.
-      val s = displayMasked(key, mask, hasZeroConfig)
-      ("Key: " + displayPedantic(key)) |:
-        parseExpected(structure, s, expected, mask)
-    }
+      // as `Zero/name`, but could be interpreted as `Zero/Zero/name`.
+      val expected = ScopedKey(
+        Resolve(structure.extra, Select(structure.current), key.key, mask)(key.scope),
+        key.key
+      )
+      parse(structure, key, mask, hasZeroConfig)(
```

Comment:
Not sure `parse` is a good name here, since you are no longer passing `s: String`.
How about `withReparsedScopeKey(...)(f)`?

Reply:
hmm, how about `parseCheck`?

or maybe just `check` seeing as the whole spec is about parsing..

Reply:
Either is fine with me.